API/projectClass/tableUtilities.m

```diff
@@ -0,0 +1,120 @@
+classdef (Abstract) tableUtilities < handle
+
+    % This class holds the common routines for handling table entries
+
+    properties
+        paramTable = table
+    end
+
+    properties (GetAccess = protected, SetAccess = private)
+        autoNameCounter = 1
+    end
+
+    properties (Dependent, SetAccess = private)
+        paramCount
+    end
+
+    methods
+
+        function count = get.paramCount(obj)
+           count = height(obj.paramTable);
+        end
+
+        function names = getNames(obj)
+            % Get a string array of the names of each of the objects
+            % defined in the class.
+            %
+            % customFiles.getNames()
+            names = obj.paramTable{:,1};  
+        end
+        
+        function addRow(obj, row)
+            % Appends a new row to the table. Expects a cell array  
+            % with the row to append
+            % 
+            % obj.addRow({'Tails', 10, 20, 30, true, 'uniform', 0, Inf})
```

Comment:
Don't need the  cell array `obj.addRow('Tails', 10, 20, 30, true, 'uniform', 0, Inf)` instead?

Reply:
The routine appends a single row by concatenation, as far as I see that requires the input to be either a cell array or table.

Reply:
You can use varargin in backgrounds and resolutions, but the addRow routines in multiTypeTable and tableUtilities are not called directly by the user.